tests/sparse_test.py

```diff
@@ -731,9 +731,13 @@ def test_bcoo_todense_partial_batch(self, shape, dtype, n_batch, n_dense):
       for lhs_shape, rhs_shape, lhs_contracting, rhs_contracting in [
           [(5,), (6,), [], []],
           [(5,), (5,), [0], [0]],
+          [(5,), (5, 7), [0], [0]],
+          [(5,), (7, 5), [0], [1]],
           [(5, 7), (5,), [0], [0]],
           [(7, 5), (5,), [1], [0]],
           [(3, 5), (2, 5), [1], [1]],
+          [(3, 5), (5, 2), [1], [0]],
+          [(5, 3), (2, 5), [0], [1]],
```

Comment:
It wouldn't hurt I think to cover these in a new GPU-only test method specifically to test cases that should be lowered to GPU. If something changes with cusparse and leads to those failing, it would be a much clearer signal of the root issue.

Reply:
Separate the test cases into a standalone test method. Is there a way to only pick up the method on GPU?

Reply:
Yes, you can do something similar to this: https://github.com/google/jax/blob/1388dbe25c3b1c003955a9f4743dca72668d2d8d/tests/sparse_test.py#L135